OneBusAway/MainPage.xaml.cs

```
@@ -186,11 +200,69 @@ protected override void OnNavigatedTo(System.Windows.Navigation.NavigationEventA
+                    Task.Run(() => InstallVoiceCommands());
+                }
+            }
+            else
+            {
+                // If we're returning to the page (e.g. from suspending in the middle of anything), restore our state
+                // back to acceptance of input.
+                //SetSearchState(SearchState.ReadyForInput);
+            }
+
             viewModel.RegisterEventHandlers(Dispatcher);
 
             navigatedAway = false;
         }
 
+        private void HandleVoiceCommand(string voiceCommandName)
+        {
+            switch (voiceCommandName)
+            {
+                case "favorites":
+                    PC.SelectedIndex = (int)MainPagePivots.Favorites;
+                    PhoneApplicationService.Current.State["MainPageSelectedPivot"] = MainPagePivots.Favorites;
+                    break;
+                case "routes":
+                    PC.SelectedIndex = (int)MainPagePivots.Routes;
+                    PhoneApplicationService.Current.State["MainPageSelectedPivot"] = MainPagePivots.Routes;
+
+                    if (NavigationContext.QueryString.ContainsKey("route"))
+                        ProcessSearch(NavigationContext.QueryString["route"]);
+
+                    break;
+                case "stops":
+                    PC.SelectedIndex = (int)MainPagePivots.Stops;
+                    PhoneApplicationService.Current.State["MainPageSelectedPivot"] = MainPagePivots.Stops;
+                    break;
+                case "recent":
+                    PC.SelectedIndex = (int)MainPagePivots.Recents;
+                    PhoneApplicationService.Current.State["MainPageSelectedPivot"] = MainPagePivots.Recents;
+                    break;
+                case "NaturalLanguage":
+                    ProcessSearch(NavigationContext.QueryString["naturalLanguage"]);
+                    break;
+                default:
+                    throw new NotSupportedException();

```

Hmm, is this the right behavior? Is it possible to bring the app down by asking for something that OBA doesn't understand? What is the problem with swallowing the exception?